plaidbench.py

```diff
@@ -80,6 +78,111 @@ def has_plaid():
         return False
 
 
+def value_check(examples, epochs, batch_size):
+    if epochs > examples:
+        raise ValueError('The number of epochs must be less than the number of examples.')
+    if batch_size > (examples // epochs):
+        raise ValueError('The number of examples per epoch must be greater than the batch size.')
+    if examples % epochs != 0:
+        raise ValueError('The number of examples must be divisible by the number of epochs.')
+    if (examples // epochs) % batch_size != 0:
+        raise ValueError('The number of examples per epoch is not divisble by the batch size.')
+
+
+def train(x_train, y_train, epoch_size, model, batch_size, compile_stop_watch, 
+    epochs, stop_watch, output, network):
+    # Training
+    compile_stop_watch.start_outer()
+    stop_watch.start_outer()
+    
+    run_intial(batch_size, compile_stop_watch, network, model)
+    x = x_train[:epoch_size]
+    y = y_train[:epoch_size]
+    model.train_on_batch(x_train[0:batch_size], y_train[0:batch_size])
+    
+    compile_stop_watch.stop()
+
+    for i in range(epochs):
+        if i == 1:
+            printf('Doing the main timing')
+        stop_watch.start()
+        history = model.fit(
+            x=x, y=y, batch_size=batch_size, epochs=1, shuffle=False, initial_epoch=0)
+        stop_watch.stop()
+        time.sleep(.025 * random.random())
+        if i == 0:
+            output.contents = [history.history['loss']]
+    output.contents = np.array(output.contents)
```

Comment:
So, are you intentionally only capturing the first loss?  It might be nice to add a comment about that; it looks wrong.